x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/deployment/InferencePyTorchAction.java

```diff
@@ -196,9 +199,13 @@ private void processResult(
             return;
         }
         if (isCancelled()) {
+            getProcessContext().getTimeoutCount().incrementAndGet();
+            getProcessContext().getResultProcessor().ignoreResponseWithoutNotifying(String.valueOf(pyTorchResult.requestId()));
             onFailure("inference task cancelled");
             return;
         }
+
+        getProcessContext().getResultProcessor().updateStats(pyTorchResult);
```

Comment:
The stats for successful requests should be updated after the cancellation check. Otherwise the request is counted as successful and cancelled, which gives incorrect total request count.

Reply:
👍